src/overlays/actors/ovl_Bg_Jya_Megami/z_bg_jya_megami.c

```diff
@@ -21,31 +37,325 @@ const ActorInit Bg_Jya_Megami_InitVars = {
+    { { -4.32f, -13.44f, -13.76f }, 0.0f, 0x04B0, 0x0190, 42 },
+    { { 30.92f, -20.480001f, -28.84f }, 1.6f, 0xFCE0, 0x0320, 44 },
+    { { -24.08f, -40.960003f, -21.359999f }, -1.0f, 0xFD44, 0x01F4, 36 },
+    { { -44.8f, -73.92f, -49.76f }, -3.2f, 0x03E8, 0xFF38, 30 },
+    { { -9.0f, -76.479996f, -13.24f }, -1.0f, 0xFC18, 0x0258, 26 },
+    { { -10.240001f, -52.56f, -13.400001f }, 1.6f, 0x0258, 0xFE70, 34 },
+    { { 34.04f, -61.72f, -37.04f }, 1.5f, 0x0258, 0x0258, 28 },
+    { { 59.48f, -38.399998f, -49.4f }, 1.8f, 0x01F4, 0x0000, 38 },
+    { { -19.04f, -112.24f, -35.120003f }, -1.6f, 0x012C, 0x0320, 22 },
+    { { 12.24f, -99.04f, -31.64f }, 1.8f, 0xFC18, 0xFE70, 24 },
+    { { 49.16f, -81.24f, -55.52f }, 2.4f, 0x02BC, 0x012C, 32 },
+    { { 14.759999f, -125.8f, -44.16f }, 0.2f, 0x0320, 0x0258, 20 },
+};
+
+// sEffectScales
+s16 D_8089B14C[] = {
+    0x0005, 0x0008, 0x000B, 0x000E, 0x0011, 0x0014, 0x0017, 0x001A,
+};
+
+// sEffectLifes
+s16 D_8089B15C[] = {
+    0x0012, 0x001A, 0x0022, 0x002A, 0x0032, 0x003C, 0x0046, 0x0050,
+};
+
+// sEffectRotSpeeds
+s16 D_8089B16C[] = {
+    0x0030, 0x002A, 0x0024, 0x0020, 0x001C, 0x0018, 0x0014, 0x0010,
+};
+
+s16 sBitMasks[] = {
+    0x0001,
+    0x0003,
+    0x0007,
+};
+
+// sBurstDepthX
+static Vec3f D_8089B184 = {
+    0.0f,
+    0.0f,
+    0.8f,
+};
+
+static InitChainEntry sInitChain[] = {
+    ICHAIN_VEC3F_DIV1000(scale, 100, ICHAIN_CONTINUE),
+    ICHAIN_F32(uncullZoneForward, 1200, ICHAIN_CONTINUE),
+    ICHAIN_F32(uncullZoneScale, 800, ICHAIN_CONTINUE),
+    ICHAIN_F32(uncullZoneDownward, 1200, ICHAIN_STOP),
+};
+
+static Vec3f sVec = {
+    0.0f,
+    0.0f,
+    0.0f,
+};
+
+static UNK_PTR sRightSideCrumbles[] = {
+    0x06000D00, 0x06001D00, 0x06002500, 0x06002D00, 0x06004D00,
+};
+
+static UNK_PTR sLeftSideCrumbles[] = {
+    0x06001500, 0x06003500, 0x06003D00, 0x06004500, 0x06000500,
+};
+
+static Gfx* sDLists[] = {
+    0x06009928, 0x06009AC0, 0x06009C80, 0x06009DE8, 0x06009F60, 0x0600A0A8, 0x0600A278,
+    0x0600A418, 0x0600A568, 0x0600A6A0, 0x0600A7E0, 0x0600A978, 0x0600AAC8,
+};
+
+void BgJyaMegami_SetupDynapoly(BgJyaMegami* this, GlobalContext* globalContext, void* collision,
+                               DynaPolyMoveFlag flags) {
+    s32 pad;
+    u32 temp;
+
+    temp = 0;
+    DynaPolyInfo_SetActorMove(&this->dyna, flags);
+    DynaPolyInfo_Alloc(collision, &temp);
+    this->dyna.dynaPolyId =
+        DynaPolyInfo_RegisterActor(globalContext, &globalContext->colCtx.dyna, &this->dyna.actor, temp);
+}
+
+void BgJyaMegami_SetupCollider(BgJyaMegami* this, GlobalContext* globalContext) {
+    s32 pad;
+
+    Collider_InitJntSph(globalContext, &this->collider);
+    Collider_SetJntSph(globalContext, &this->collider, &this->dyna.actor, &sJntSphInit, &this->colliderItem);
+}
+
+void func_8089A1DC(GlobalContext* globalContext, Vec3f* burstOrigin, Vec3f* burstDepthX, s32 num, s32 maskIndex) {
+    Vec3f spB4;
+    s32 i;
+
+    for (i = 0; i < num; ++i) {
```

Comment:
Avoid pre-incrementing unless necessary, for consistency with the rest of the codebase. (relevant throughout the file)
```suggestion
    for (i = 0; i < num; i++) {
```